test/e2e/common/podtemplates.go

```diff
@@ -99,4 +100,79 @@ var _ = ginkgo.Describe("[sig-node] PodTemplates", func() {
 		framework.ExpectNoError(err, "failed to list PodTemplate")
 		framework.ExpectEqual(len(podTemplateList.Items), 0, "PodTemplate list returned items, failed to delete PodTemplate")
 	})
+
+	ginkgo.It("should delete a collection of pod templates", func() {
+		podTemplateNames := []string{"test-podtemplate-1", "test-podtemplate-2", "test-podtemplate-3"}
+		podTemplateNamesCount := len(podTemplateNames)
+
+		ginkgo.By("Create set of pod templates")
+		// create a set of pod templates in test namespace
+		for _, podTemplateName := range podTemplateNames {
+			_, err := f.ClientSet.CoreV1().PodTemplates(f.Namespace.Name).Create(context.TODO(), &v1.PodTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   podTemplateName,
+					Labels: map[string]string{"podtemplate-set": "true"},
+				},
+				Template: v1.PodTemplateSpec{
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{Name: "nginx", Image: "nginx"},
+						},
+					},
+				},
+			}, metav1.CreateOptions{})
+			framework.ExpectNoError(err, "failed to create pod template")
+			framework.Logf("created %v", podTemplateName)
+		}
+
+		ginkgo.By("get a list of pod templates with a label in the current namespace")
+		// get a list of pod templates
+		podTemplateList, err := f.ClientSet.CoreV1().PodTemplates(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "podtemplate-set=true",
+		})
+		framework.ExpectNoError(err, "failed to get a list of pod templates")
+
+		// check that we find all the pod templates created
+		podTemplateListItemsCount := len(podTemplateList.Items)
+		errMsg := "found " + strconv.Itoa(podTemplateListItemsCount) + " pod templates when " + strconv.Itoa(podTemplateNamesCount) + " pod templates where expected"
+		logMsg := "found " + strconv.Itoa(podTemplateListItemsCount) + " pod templates"
+
+		foundCreatedSet := true
+		if podTemplateListItemsCount != podTemplateNamesCount {
+			foundCreatedSet = false
+		} else {
+			framework.Logf(logMsg)
+		}
+		framework.ExpectEqual(foundCreatedSet, true, errMsg)
```

Comment:
I'm not sure we need 11 lines to express when 1 would do? They make it harder to understand what this test is doing at a glance
```suggestion
		framework.ExpectEqual(len(podTemplateList.Items), len(podTemplateNamesCount), "looking for expected number of pod templates")
```

Reply:
[Fixed](https://github.com/kubernetes/kubernetes/commit/758e3c76a4fd8b037948713f032f47545f1b0664)